src/components/Cropper.js

```diff
@@ -325,22 +339,60 @@ class Cropper extends Component {
     }
   }
 
+  /**
+   * To prevent picture to be out of the hole
+   * We need to check it to replace picture into the hole
+   * @param {Number} newZoom
+   * @param {Number} newRotation
+   */
+  _fixPicturePositionOnZoom(newZoom, newRotation = false) {
+    let {
+      picturePositionX,
+      picturePositionY,
+      currentZoom,
+      pictureHeight,
+      pictureWidth,
+      currentRotation,
+    } = this.state
+    const {
+      holeSize,
+    } = this.props
+
+    newRotation = newRotation !== false ? newRotation : currentRotation
+
+    // Zoom on picture center
+    picturePositionX = (picturePositionX / currentZoom * newZoom)
+    picturePositionY = (picturePositionY / currentZoom * newZoom)
+
+    // fix picture out of bounds
+    const bottomBoundsPictureMarge = (((newRotation / 90) % 2 === 0 ? pictureHeight : pictureWidth) * newZoom) - picturePositionY - holeSize
```

Comment:
Ca vaudrait pas le coup de stocker une variable de state `position: 'landscape' ou 'portrait' comme ça on peut faire this.state.position === 'portrait'` (voire carrément une méthode `isLandscape` ?)

Le but étant d'écrire des trucs intuitifs parce que `(newRotation / 90) % 2 === 0` j'ai buggé pdt un moment et encore je suis pas entièrement convaincu que ce soit portrait/landscape. (pour l'instant le component n'autorise que les rotations multiples de 90°, mais du coup j'aurais isolé ça dans une méthode `isPortrait/isLandscape`)